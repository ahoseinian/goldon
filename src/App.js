import React, {Component} from 'react';
import TopBar from './components/TopBar'
import Footer from './components/footer/Footer'
import styled from 'styled-components'

const App = ({pathname, children}) => (
  <div>
    <TopBar pathname={pathname}/> {children}
    <Footer/>
  </div>
)

App.propTypes = {
  pathname: React.PropTypes.string.isRequired,
  children: React.PropTypes.oneOfType([
    React.PropTypes.arrayOf(React.PropTypes.node),
    React.PropTypes.node
  ])
}

export default App
