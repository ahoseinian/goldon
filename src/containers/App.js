import React from 'react';
import TopBar from '../components/TopBar'
import Footer from '../components/footer/Footer'

const App = ({location, children}) => (
  <div>
    <TopBar pathname={location.pathname}/>
    <div style={{marginTop:'3.5rem'}}>
      {children}
    </div>
    <Footer/>
  </div>
)

App.propTypes = {
  location: React.PropTypes.object.isRequired,
  children: React.PropTypes.oneOfType([
    React.PropTypes.arrayOf(React.PropTypes.node),
    React.PropTypes.node
  ])
}

export default App
