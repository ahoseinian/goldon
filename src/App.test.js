import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import App from './App'

describe('App', () => {
  it('renders with children', () => {
    const wrapper = shallow(<App pathname="/">test</App>)
    expect(toJson(
      wrapper
    )).toMatchSnapshot()
  })
})
