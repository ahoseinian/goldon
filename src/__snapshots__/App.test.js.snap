exports[`App renders with children 1`] = `
<div>
  <TopBar
    pathname="/" />
   
  test
  <Footer />
</div>
`;
